src/index.ts

```diff
@@ -87,7 +78,7 @@ export function useMappedState<TState, TResult>(
         unsubscribe();
       };
     },
-    [store, mapState],
+    [store, ...memoizationArray],
```

Comment:
I don't think we want this here, as it would cause the store to re-render every time one of the inputs change (although it already does that for mapState, it should just depend on mapState explicitly)

Reply:
So your idea is to wrap mapState in a useCallback and use that value for the useEffect?

Reply:
Yup!

Reply:
I was thinking about maybe creating a memorized wrapper function within the hook such that you never need to resubscribe (unless the store changes) and this way it might be possible to ensure the top-down subscription order.